src/compas_rhino/geometry/curves/nurbs.py

```diff
@@ -0,0 +1,527 @@
+        curve.rhino_curve = Rhino.Geometry.NurbsCurve.Create(is_periodic, degree, points)
+        return curve
+
+    @classmethod
+    def from_interpolation(cls, points, precision=1e-3):
+        """Construct a NURBS curve by interpolating a set of points.
+
+        This construction method is similar to the method ``CreateHSpline`` of the Rhino API for NURBS curves [1]_.
+
+        References
+        ----------
+        .. [1] https://developer.rhino3d.com/api/RhinoCommon/html/Overload_Rhino_Geometry_NurbsCurve_CreateHSpline.htm
+        .. [2] https://dev.opencascade.org/doc/occt-7.4.0/refman/html/class_geom_a_p_i___interpolate.html
+
+        """
+        curve = cls()
+        curve.rhino_curve = Rhino.Geometry.NurbsCurve.CreateHSpline([point_to_rhino(point) for point in points])
+        return curve
+
+    @classmethod
+    def from_step(cls, filepath):
+        """Load a NURBS curve from an STP file."""
+        pass
+
+    @classmethod
+    def from_arc(cls, arc, degree, pointcount=None):
+        pass
+
+    @classmethod
+    def from_circle(cls, circle):
+        """Construct a NURBS curve from a circle.
+
+        This construction method is similar to the method ``CreateFromCircle`` of the Rhino API for NURBS curves [1]_.
+
+        References
+        ----------
+        .. [1] https://developer.rhino3d.com/api/RhinoCommon/html/Overload_Rhino_Geometry_NurbsCurve_CreateFromCircle.htm
+
+        """
+        curve = cls()
+        curve.rhino_curve = Rhino.Geometry.NurbsCurve.CreateFromCircle(circle_to_rhino(circle))
+        return curve
+
+    @classmethod
+    def from_ellipse(cls, ellipse):
+        """Construct a NURBS curve from an ellipse.
+
+        This construction method is similar to the method ``CreateFromEllipse`` of the Rhino API for NURBS curves [1]_.
+
+        References
+        ----------
+        .. [1] https://developer.rhino3d.com/api/RhinoCommon/html/Overload_Rhino_Geometry_NurbsCurve_CreateFromEllipse.htm
+
+        """
+        curve = cls()
+        curve.rhino_curve = Rhino.Geometry.NurbsCurve.CreateFromEllipse(ellipse_to_rhino(ellipse))
+        return curve
+
+    @classmethod
+    def from_line(cls, line):
+        """Construct a NURBS curve from a line.
+
+        This construction method is similar to the method ``CreateFromLine`` of the Rhino API for NURBS curves [1]_.
+
+        References
+        ----------
+        .. [1] https://developer.rhino3d.com/api/RhinoCommon/html/Overload_Rhino_Geometry_NurbsCurve_CreateFromLine.htm
+
+        """
+        curve = cls()
+        curve.rhino_curve = Rhino.Geometry.NurbsCurve.CreateFromLine(line_to_rhino(line))
+        return curve
+
+    # ==============================================================================
+    # Conversions
+    # ==============================================================================
+
+    def to_step(self, filepath, schema="AP203"):
+        """Write the curve geometry to a STP file."""
+        pass
```

Comment:
These 3 methods as well, i guess they need to raise an exception instead of passing